bindings/gcp/bucket/bucket.go

```diff
@@ -54,19 +54,21 @@ type GCPStorage struct {
 }
 
 type gcpMetadata struct {
-	Bucket              string `json:"bucket" mapstructure:"bucket"`
-	Type                string `json:"type" mapstructure:"type"`
-	ProjectID           string `json:"project_id" mapstructure:"project_id"`
-	PrivateKeyID        string `json:"private_key_id" mapstructure:"private_key_id"`
-	PrivateKey          string `json:"private_key" mapstructure:"private_key"`
-	ClientEmail         string `json:"client_email " mapstructure:"client_email"`
-	ClientID            string `json:"client_id" mapstructure:"client_id"`
-	AuthURI             string `json:"auth_uri" mapstructure:"auth_uri"`
-	TokenURI            string `json:"token_uri" mapstructure:"token_uri"`
-	AuthProviderCertURL string `json:"auth_provider_x509_cert_url" mapstructure:"auth_provider_x509_cert_url"`
-	ClientCertURL       string `json:"client_x509_cert_url" mapstructure:"client_x509_cert_url"`
-	DecodeBase64        bool   `json:"decodeBase64,string" mapstructure:"decodeBase64"`
-	EncodeBase64        bool   `json:"encodeBase64,string" mapstructure:"encodeBase64"`
+	// Ignored by metadata parser because included in built-in authentication profile
+	Type                string `json:"type" mapstructure:"type" mdignore:"true"`
+	ProjectID           string `json:"project_id" mapstructure:"projectID" mdignore:"true" mapstructurealiases:"project_id"`
```

Comment:
For this field and all the fields below please **either**:

- Remove the `json` struct tag
- OR Make sure that it's the same as `mapstructure` (not the alias)

Reply:
I'm going to check if the [option.WithCredentialsJSON(b)](https://github.com/dapr/components-contrib/blob/master/bindings/gcp/bucket/bucket.go#L99) requires those `json` struct tags.